### ISSUE TITLE
Option 'iminsert' isn't a local to buffer as it should be

### ISSUE THREAD

**Issue by dtcyganok** [vim/vim#2390]
### Steps to reproduce

```
vim -u NORC any-file.txt
:setlocal imiminsert=1
:e another-file.txt
:setlocal imiminsert?
```

**Comment by bfredl** [comment]
Related issue: https://github.com/neovim/neovim/issues/7658

The behavior is caused by this line:
https://github.com/vim/vim/blob/46359e198f6d6884dc3d3c4a3e46625f1b2a2ad2/src/option.c#L8915

Not sure why it is copied back and forth like that. Only `b_p_iminsert` ever takes effect, not `p_iminsert`.

**Comment by chrisbra** [comment]
Yeah, I have mentioned that before: https://groups.google.com/d/msg/vim_dev/HpwZWJojl3E/ndWKRSYaUXcJ

**Comment by brammool** [comment]
Dmitrii Tcyganok wrote:

> ### Steps to reproduce
> 
> ```
> vim -u NORC any-file.txt
> :setlocal imiminsert=1
> :e another-file.txt
> :setlocal imiminsert?
> ```

There are several places where the buffer-local value is copied to the
global value.  The value is only used when entering a new buffer, it
seems.

I don't use an input method myself, but I would think that not all that
much related to the buffer.  Perhaps a global-local option would have
been better.  But changing that won't be backwards compatible.

-- 
Over the years, I've developed my sense of deja vu so acutely that now
I can remember things that *have* happened before ...

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///

**Comment by dtcyganok** [comment]
I constantly use 'iminsert'. I find the current behavior of the option confusing and uncomfortable.
Currently i use this in my vimrc:
```vim
autocmd BufNewFile,BufRead * setlocal iminsert=0
```
As a vim user, i would prefer usability over backward compatibility, if it's possible.
Are there any specific problems of backwards compatibility?
